Automattic/jetpack
# Jetpack AI: Add renewal date to fair usage messaging

## Proposed changes:

* Add hook to handle fair usage messaging creation
* Add new message with usage renewal date and link to usage policy
* Format date to be readable, like "September 22"
* Change notice status from "warning" to "error" so it is red, as per proposed design

### Other information:


## Jetpack product discussion



p1HpG7-tUE-p2

## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Run this branch on your testing site
* Sandbox the `public-api` domain and change the `0-sandbox.php` file to add the following filters, so we simulate an unlimited plan over the fair usage limit:

```
add_filter( 'jetpack_ai_tier_licensed_quantity', function() { return 1; } );
add_filter( 'jetpack_ai_current_period_requests_count', function() { return 3001; } );
```

* Go to the block editor and add an AI Assistant block
* Confirm you see the new version of the fair usage notice, containing the reset date, a link to the fair usage policy (actually, a Jetpack redirect link still not pointing to a final page) and red background:

<img width="500" alt="Screenshot 2024-08-27 at 12 09 49" src="https://github.com/user-attachments/assets/9e2dbcbc-a7b8-4d37-b21c-83251dffc10c">

* Insert an image block and click "Generate with AI"
* Confirm you also see the new version of the fair usage notice

<img width="500" alt="Screenshot 2024-08-27 at 12 09 58" src="https://github.com/user-attachments/assets/8aa530da-433e-439d-a89e-04463442a32a">

* Last, check the "Excerpt" section on the post sidebar and confirm the new fair usage notice is also there:

<img width="250" alt="Screenshot 2024-08-27 at 12 10 14" src="https://github.com/user-attachments/assets/980327a7-eab6-4888-bb86-18f005c0f387">
